packages/next/src/build/webpack/plugins/next-trace-entrypoints-plugin.ts

```diff
@@ -664,6 +693,29 @@ export class TraceEntryPointsPlugin implements webpack.WebpackPluginInstance {
                       }
                     }
                   }
+
+                  // ensure we grab all associated dependencies
+                  if (this.flyingShuttle) {
+                    const curDepSet = entryPathDepMap.get(entry)
+
+                    for (const item of curDepSet || []) {
+                      if (!item.includes('?')) {
+                        if (!finalDeps.has(item)) {
+                          finalDeps.set(item, {
+                            bundled: true,
+                          })
+                        }
+                        for (const [dep, info] of parentFilesMap
+                          .get(nodePath.relative(this.tracingRoot, entry))
+                          ?.entries() || []) {
```

Comment:
maybe we can hoist this `of` part into a variable (`parentEntries` or something?), it's a bit hard to read right now

Reply:
updated